packages/api/src/web3Middleware/index.ts

```diff
@@ -127,22 +139,25 @@ export class Web3Middleware {
   }
 
   async listenToDataFeed (feedInfo: FeedInfo) {
-    const contractAddress = await this.getContractAddress(feedInfo)
+    const contractInfo = await this.getContractAddress(feedInfo)
     const provider = getProvider(feedInfo.network)
     if (provider) {
+      console.log('contractInfo in listenToDataFeed', contractInfo)
       if (
-        contractAddress &&
-        contractAddress !== '0x0000000000000000000000000000000000000000'
+        contractInfo &&
+        contractInfo.contractAddress &&
+        contractInfo.contractAddress !==
+          '0x0000000000000000000000000000000000000000'
```

Comment:
Can we move this to a utils function?

```
function isZeroAddress(address: string) {
  return address === '0x0000000000000000000000000000000000000000'
}
```

```suggestion
        contractInfo.contractAddress &&
        !isZeroAddress(contractInfo.contractAddress)
```